analyses/malawi/flooding_trigger/04_glofas_skill.md

```diff
@@ -0,0 +1,183 @@
+
+```python
+def plot_crps(df_crps, title_suffix=None, ylog=False):
+    for code, station in stations_adm2.items():
+        fig, ax = plt.subplots()
+        crps = df_crps[code]
+        ax.plot(crps.index, crps, label=station)
+        ax.legend()
+        title = station
+        if title_suffix is not None:
+            title += title_suffix
+        ax.set_title(title)
+        ax.set_xlabel("Lead time [days]")
+        ax.set_ylabel("Normalized CRPS [% error]")
+        ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+        ax.grid()
+        if ylog:
+            ax.set_yscale('log')
+            ax.yaxis.set_major_formatter(ScalarFormatter())
+```
+
+```python
+df_crps = utils.get_crps(ds_glofas_reanalysis, 
+                         ds_glofas_reforecast,
+                        normalization="mean")
+plot_crps(df_crps * 100, title_suffix=" -- all discharge values")
+
+if SAVE_FIG: plt.savefig(PLOT_DIR / f'{station}_ncrps_all.png')
+```
+
+```python
+rp = 3
+df_crps = utils.get_crps(ds_glofas_reanalysis, 
+                         ds_glofas_reforecast,
+                         normalization="mean", 
+                         thresh=df_return_period.loc[rp].to_dict())
+plot_crps(df_crps * 100, title_suffix=f" -- values > RP 1 in {rp} y", ylog=False)
+
+if SAVE_FIG: plt.savefig(PLOT_DIR / f'{station}_ncrps_{rp}_rp.png')
+```
+
+We'll also look at the forecast bias, again specifically at extreme water discharge levels. 
+
+```python
+# Rank histogram
+def plot_hist(da_observations, da_forecast, station_name, rp=None, leadtimes=None):
+    if leadtimes is None:
+        leadtime = da_forecast.leadtime.values
+    fig, ax = plt.subplots()
+    for leadtime in leadtimes:
+        observations, forecast = utils.get_same_obs_and_forecast(da_observations, da_forecast, leadtime)
+        rank = utils.get_rank(observations.values, forecast.values)
+        ax.hist(rank, histtype='step', label=int(leadtime),
+               bins=np.arange(0.5, max(rank)+1.5, 1), alpha=0.8)
+    ax.legend(loc=9, title="Lead time (days)")
+    ax.set_xlabel('Rank')
+    ax.set_ylabel('Number')
+    title = station_name
+    if rp is not None:
+        title += f': > 1 in {rp} y'
+    ax.set_title(title)
+
+rp = 3
+leadtimes = [5, 10, 15, 20]
+for code, station in stations_adm2.items():
+    da_observations =  ds_glofas_reanalysis[code]
+    da_forecast = ds_glofas_reforecast[code]
+    plot_hist(da_observations, da_forecast, station, leadtimes=[5, 10, 15, 20])
+    
+    if SAVE_FIG: 
+        plt.savefig(PLOT_DIR / f'{station}_rank_hist_all.png')
+    
+    rp_val = df_return_period.loc[rp, code]
+    o = da_observations[da_observations > rp_val]
+    # Needs at least about 50 vals to work, not sure why
+    if len(o) > 50:
+        plot_hist(o, da_forecast, station, leadtimes=leadtimes, rp=rp)
+        
+        if SAVE_FIG: plt.savefig(PLOT_DIR / f'{station}_rank_hist_{rp}_rp.png')
+```
+
+```python
+rp = 2
```

Comment:
Why do you use rp = 2 here but 3 elsewhere? Maybe it's better to set a global rp var?